src/base/rss/rss_autodownloader.cpp

```diff
@@ -52,6 +52,36 @@
 #include "rss_folder.h"
 #include "rss_session.h"
 
+namespace {
+    // find all occurrences of <placeholder> in original string, and replace them with value of "placeholder" from article
+    // example: savePathTpl = "c:\downloads\<attr_name> - <description>\something\<title>\<not_found_attr>"
+    //          article = <item>
+    //                      <title>some title</title>
+    //                      <link>some link</link>
+    //                      <guid isPermaLink="false">some-guid</guid>
+    //                      <pubDate>Thu, 02 Jun 2016 03:50:04 +0000</pubDate>
+    //                      <description>some description</description>
+    //                      <attr_name>some attr value</attr_name>
+    //                    </item>
+    //          result = "c:\downloads\some attr value - some description\something\some title\not_found_attr"
+    QString generateSavePath(const QString savePath, const QVariantHash articleData)
+    {
+        QRegExp rx("<(.+)>");
```

Comment:
Don't like this syntax because it is hardly extensible. For example, how would you add default values? One has to reserve a set of characters for the special purpose like denoting a default value or value prefixes/affixes. If we release this free form into the wild life, users create directory structures with all possible characters in names and we will have problems in the future extending the rule syntax.